In [1]:
import pandas as pd

In [5]:
df=pd.read_csv(r"D:\Bluestock\Project\Data\Raw\01_fund_master.csv")

In [6]:
df.head()

,amfi_code,fund_house,scheme_name,category,sub_category,plan,launch_date,benchmark,expense_ratio_pct,exit_load_pct,min_sip_amount,min_lumpsum_amount,fund_manager,risk_category,sebi_category_code
0,119551,SBI Mutual Fund,SBI Bluechip Fund - Regular Plan - Growth,Equity,Large Cap,Regular,2006-02-14,NIFTY 100 TRI,1.54,1.0,500,1000,Sohini Andani,Moderate,EC01
1,119552,SBI Mutual Fund,SBI Bluechip Fund - Direct Plan - Growth,Equity,Large Cap,Direct,2013-01-01,NIFTY 100 TRI,0.66,1.0,500,1000,Sohini Andani,Moderate,EC01
2,119598,SBI Mutual Fund,SBI Small Cap Fund - Regular Plan - Growth,Equity,Small Cap,Regular,2009-09-09,BSE 250 SmallCap TRI,1.43,1.0,500,1000,R. Srinivasan,Very High,EC03
3,119599,SBI Mutual Fund,SBI Small Cap Fund - Direct Plan - Growth,Equity,Small Cap,Direct,2013-01-01,BSE 250 SmallCap TRI,0.72,1.0,500,1000,R. Srinivasan,Very High,EC03
4,119120,SBI Mutual Fund,SBI Magnum Gilt Fund - Regular Plan - Growth,Debt,Gilt,Regular,2000-12-30,CRISIL Dynamic Gilt Index,0.77,0.0,500,1000,Dinesh Ahuja,Low,DC02


In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   amfi_code           40 non-null     int64  
 1   fund_house          40 non-null     str    
 2   scheme_name         40 non-null     str    
 3   category            40 non-null     str    
 4   sub_category        40 non-null     str    
 5   plan                40 non-null     str    
 6   launch_date         40 non-null     str    
 7   benchmark           40 non-null     str    
 8   expense_ratio_pct   40 non-null     float64
 9   exit_load_pct       40 non-null     float64
 10  min_sip_amount      40 non-null     int64  
 11  min_lumpsum_amount  40 non-null     int64  
 12  fund_manager        40 non-null     str    
 13  risk_category       40 non-null     str    
 14  sebi_category_code  40 non-null     str    
dtypes: float64(2), int64(3), str(10)
memory usage: 4.8 KB


In [8]:
df.describe()

,amfi_code,expense_ratio_pct,exit_load_pct,min_sip_amount,min_lumpsum_amount
count,40.000000,40.000000,40.000000,40.0,40.000000
mean,120247.000000,1.237000,0.812500,500.0,1277.500000
std,14534.998667,0.386584,0.387091,0.0,1082.847031
min,100016.000000,0.550000,0.000000,500.0,100.000000
25%,118632.750000,0.787500,1.000000,500.0,1000.000000
50%,119551.500000,1.425000,1.000000,500.0,1000.000000
75%,120842.250000,1.540000,1.000000,500.0,1000.000000
max,149324.000000,1.640000,1.000000,500.0,5000.000000


In [9]:
df["fund_house"].unique()

<StringArray>
[         'SBI Mutual Fund',         'HDFC Mutual Fund',
      'ICICI Prudential MF',          'Nippon India MF',
        'Kotak Mahindra MF',         'Axis Mutual Fund',
 'Aditya Birla Sun Life MF',          'UTI Mutual Fund',
           'Mirae Asset MF',          'DSP Mutual Fund']
Length: 10, dtype: str

In [14]:
df["fund_house"].value_counts()

fund_house
SBI Mutual Fund             5
HDFC Mutual Fund            5
ICICI Prudential MF         5
Nippon India MF             5
Kotak Mahindra MF           4
Axis Mutual Fund            4
Aditya Birla Sun Life MF    3
UTI Mutual Fund             3
Mirae Asset MF              3
DSP Mutual Fund             3
Name: count, dtype: int64

In [10]:
df["category"].unique()

<StringArray>
['Equity', 'Debt']
Length: 2, dtype: str

In [15]:
df["category"].value_counts()

category
Equity    34
Debt       6
Name: count, dtype: int64

In [11]:
df["sub_category"].unique()

<StringArray>
[      'Large Cap',       'Small Cap',            'Gilt',         'Mid Cap',
  'Short Duration',           'Value',          'Liquid',       'Index/ETF',
       'Flexi Cap',           'Index', 'Large & Mid Cap',            'ELSS']
Length: 12, dtype: str

In [16]:
df["sub_category"].value_counts()

sub_category
Large Cap          14
Mid Cap             7
Small Cap           6
Liquid              3
Gilt                2
Flexi Cap           2
Short Duration      1
Value               1
Index/ETF           1
Index               1
Large & Mid Cap     1
ELSS                1
Name: count, dtype: int64

In [12]:
df["risk_category"].unique()

<StringArray>
['Moderate', 'Very High', 'Low', 'High', 'Moderately High']
Length: 5, dtype: str

In [17]:
df["risk_category"].value_counts()

risk_category
Moderate           16
High                8
Very High           6
Low                 6
Moderately High     4
Name: count, dtype: int64

In [18]:
df[["amfi_code", "scheme_name"]].head(10)

,amfi_code,scheme_name
0,119551,SBI Bluechip Fund - Regular Plan - Growth
1,119552,SBI Bluechip Fund - Direct Plan - Growth
2,119598,SBI Small Cap Fund - Regular Plan - Growth
3,119599,SBI Small Cap Fund - Direct Plan - Growth
4,119120,SBI Magnum Gilt Fund - Regular Plan - Growth
5,100016,HDFC Top 100 Fund - Regular Plan - Growth
6,125497,HDFC Top 100 Fund - Direct Plan - Growth
7,100033,HDFC Mid-Cap Opportunities Fund - Regular - Gr...
8,125498,HDFC Mid-Cap Opportunities Fund - Direct - Growth
9,100025,HDFC Short Term Debt Fund - Regular - Growth


# Understanding AMFI Scheme Codes

## Observations

- Each mutual fund scheme is assigned a unique AMFI code.
- Direct and Regular plans of the same fund have different AMFI codes.
  - Example:
    - SBI Bluechip Fund - Regular Plan → 119551
    - SBI Bluechip Fund - Direct Plan → 119552
- AMFI codes are not necessarily sequential across all schemes.
  - Example:
    - HDFC Top 100 Fund - Regular Plan → 100016
    - HDFC Top 100 Fund - Direct Plan → 125497
- The AMFI code serves as a unique identifier for every scheme-plan combination.

## Role in the Project

AMFI codes are used to:

- Link Fund Master and NAV History datasets.
- Validate referential integrity across datasets.
- Fetch scheme data from MFAPI endpoints.
- Support ETL processing, analytics, and dashboard development.

## Conclusion

The AMFI code is the primary key used throughout the Mutual Fund Analytics project. It uniquely identifies each mutual fund scheme and enables reliable integration of data across multiple sources.

# Fund Master Dataset Analysis

## Dataset Overview

- The dataset contains **40 mutual fund schemes** and **15 attributes**.
- Each scheme is uniquely identified using an **AMFI code**.
- The dataset includes important scheme-level information such as fund house, category, sub-category, benchmark, expense ratio, fund manager, and risk category.

## Key Findings

### Fund Houses
- Multiple Asset Management Companies (AMCs) are represented in the dataset.
- Major fund houses include SBI Mutual Fund, HDFC Mutual Fund, ICICI Prudential Mutual Fund, Nippon India Mutual Fund, Axis Mutual Fund, and Kotak Mutual Fund.

### Categories and Sub-Categories
- Schemes are distributed across major categories such as **Equity, Debt, and Hybrid**.
- Sub-categories include investment styles such as **Large Cap, Mid Cap, Small Cap, Flexi Cap, Gilt, Liquid, Aggressive Hybrid**, and others.

### Risk Classification
- The dataset contains schemes across different risk levels:
  - Low
  - Moderate
  - High
  - Very High

### Data Quality Observations
- No missing values were found in the dataset.
- No duplicate records were identified.
- Date fields such as `launch_date` are currently stored as object/string datatype and may require conversion to datetime during ETL processing.

## Conclusion

The Fund Master dataset serves as the primary reference table for the Mutual Fund Analytics project. It provides scheme metadata and AMFI codes that will be used to validate NAV history data and support subsequent ETL, analytics, and dashboard development. 